main.go

```diff
@@ -348,6 +334,60 @@ forloop:
 	}
 }
 
+func checkCerts() (bool, error) {
+	log.Info("Checking old certs...")
+
+	secrets := []string{
+		defaults.CiliumEtcdClientTLS,
+		defaults.CiliumEtcdServerTLS,
+		defaults.CiliumEtcdPeerTLS,
+		defaults.CiliumEtcdSecrets,
+	}
+	for _, secret := range secrets {
+		_, err := k8s.Client().CoreV1().Secrets(namespace).Get(secret, meta_v1.GetOptions{})
+		switch {
```

Comment:
Also, one of the main reasons to delete all certificates was because the certificates depend on the cluster domain. If the cluster domain changes between deployments the authentication will fail. But this is really unlikely to happen.